File: tests/exercises/echo/solution/correct-files.cs
using System;
using System.IO;

string input = Console.ReadLine();
if (File.Exists("input2.txt")) {
    string fileTxt = File.ReadAllText("input2.txt").Trim();
    if (!string.IsNullOrEmpty(fileTxt)) {
        Console.WriteLine(fileTxt);
    } else {
        Console.WriteLine(input);
    }
} else {
    Console.WriteLine(input);
}

File: tests/exercises/echo/solution/correct-files.ts
import * as fs from 'fs';

const input = fs.readFileSync(0, 'utf8').split('\n')[0];

if (fs.existsSync('input2.txt')) {
    const fileTxt = fs.readFileSync('input2.txt', 'utf8').trim();
    if (fileTxt) {
        console.log(fileTxt);
    } else {
        console.log(input);
    }
} else {
    console.log(input);
}
